pilot/cmd/pilot-agent/status/server.go

```diff
@@ -714,6 +728,66 @@ func (s *Server) handleAppProbeTCPSocket(w http.ResponseWriter, prober *Prober)
 	}
 }
 
+func (s *Server) handleAppProbeGRPC(w http.ResponseWriter, req *http.Request, prober *Prober) {
+	opts := []grpc.DialOption{
+		grpc.WithBlock(),
+		// nolint: staticcheck
+		grpc.WithInsecure(), // credentials are currently not supported
+	}
+	if userAgent := req.Header["User-Agent"]; len(userAgent) > 0 {
```

Comment:
Where is the header come from?

Reply:
It's from [kubelet](https://github.com/kubernetes/kubernetes/blob/v1.23.1/pkg/probe/http/http.go#L103)

Reply:
IC